crypto/bio/bio_mem.c

```diff
@@ -65,11 +65,18 @@
 
 #include "../internal.h"
 
+typedef struct bio_buf_mem_st {
+  struct buf_mem_st *buf;   /* allocated buffer */
+  size_t read_off;          /* read pointer offset from current buffer position */
+} BIO_BUF_MEM;
+
 
 BIO *BIO_new_mem_buf(const void *buf, ossl_ssize_t len) {
   BIO *ret;
   BUF_MEM *b;
-  const size_t size = len < 0 ? strlen((char *)buf) : (size_t)len;
+  BIO_BUF_MEM *bbm;
+
+  const size_t size = (len < 0 || (size_t)len > SIZE_MAX) ? strlen((char *)buf) : (size_t)len;
```

Comment:
I'm not familiar with these APIs but the use of `strlen` here seems wrong? Is buffer really expected to be a null terminated value?

Reply:
Apparently, this is ancient implementation from both OpenSSL and BoringSSL. Per [OpenSSL's documentation](https://docs.openssl.org/3.0/man3/BIO_s_mem/#description),

> BIO_new_mem_buf() creates a memory BIO using len bytes of data at buf, if len is -1 then the buf is assumed to be null terminated and its length is determined by strlen.

But if we don't have that assumption, I can't really think of a way to determine if `buf` is null terminated, especially since we don't have its size.